File: trunk/src/pann/Core/Link.cpp
//Link.cpp

#include "Weight.h"
#include "Neuron.h"

#include "Link.h"

using namespace boost;

namespace pann
{
    Link::Link(shared_ptr<Neuron> _to, shared_ptr<Weight> _w, unsigned const _latency) 
        : to(_to), weight(_w), latency(_latency)
    {
        weight->incUsageCount();
    } //Link

    Link::Link(const Link& _rhs)
        : to(_rhs.to), weight(_rhs.weight), latency(_rhs.latency)
    {
        weight->incUsageCount();
    } //Link

    Link::~Link()
    {
        weight->decUsageCount();
    } //~Link

/*
    Link&
    Link::operator=(Link const & _rhs)
    {
        to = _rhs.to;
        weight = _rhs.weight;
        latency = _rhs.latency;
        std::cout<<"=";

        return *this;
    } //operator=
*/

    shared_ptr<Neuron>
    Link::getTo()
    {
        if(!to)
            throw E<Exception::ObjectNotFound>()<<"Link::getTo(): Not connected\n";

        return to;
    } //getTo

    const shared_ptr<Neuron>
    Link::getTo() const
    {
        if(!to)
            throw E<Exception::ObjectNotFound>()<<"Link::getTo(): Not connected\n";

        return to;
    } //getTo

    shared_ptr<Weight>
    Link::getWeight()
    {
        if(!weight)
            throw E<Exception::ObjectNotFound>()<<"Link::getTo(): No weight\n";

        return weight;
    } //getWeight

    const shared_ptr<Weight>
    Link::getWeight() const
    {
        if(!weight)
            throw E<Exception::ObjectNotFound>()<<"Link::getTo(): No weight\n";

        return weight;
    } //getWeight

    unsigned 
    Link::getLatency() const
    {
        return latency;
    } //getLatency

}; //pann

File: src/pypann/Core/Link.py
"""
Link
"""

from Weight import Weight

class Link:
    def __init__(self, to, w, latency):
        self._to = to
        self._w = w
        self._latency = latency

        if isinstance(w, Weight):
            w.inc_usage()

    def get_to(self):
        return self._to

    def get_w(self):
        return self._w

    def get_latency(self):
        return self._latency

"""
Testing
"""
def test_Link():
    print "Testing Link..."
    l = Link(None, Weight(), 1)
    print l.get_to(), l.get_w(), l.get_latency()

if __name__ == "__main__":
    test_Link()

